engine/runtime-compiler/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -69,7 +70,6 @@ class Compiler(
     if (config.outputRedirect.isDefined)
       new PrintStream(config.outputRedirect.get)
     else context.getOut
-  private lazy val ensoCompiler: EnsoParser = new EnsoParser()
```

Comment:
Looks like the original code was buggy. The `Compiler` should have implemented `Closable` interface and call `ensoCompiler.close()` on the `EnsoParser`. I am not sure what's the overhead of a single `EnsoParser` instance and its natively allocated memory, but this must have leaked a lot during unit test execution!
- #10793

@hubertp did reduce the memory, but the Rust parser native part might now even have been visible in the JVM heap dump - only in RSS...